R/vol-surface-class.R

```diff
@@ -0,0 +1,160 @@
+#' VolSurface class
+#'
+#' The `VolSurface`` class is designed to represent volatility surfaces.
+#' Checks whether the dimentions of the surface correspond to the right type of axis: moneyness, delta or strike
+#'
+#' @param reference_date `Date` value that represents the as of date
+#' @param surface_type type of the vol dimentions. Available types: "strike/term"
+#' @param vol_quotes `matrix` of class `numeric` holding the vol values.
+#' @param dim1 `numeric` vector containing the surface strikes, deltas or
+#' moneyness depending on the surface type.
+#' @param dim2 `Date` vector containing the tenors of the surface.
+#' @param ticker label for the vol object, this should be the ticker of the underlying index
+#' @param interpolation object of type interpolation [fmbasics::Interpolation()]
+#' @export
+
+VolSurface <- function(reference_date, surface_type, vol_quotes, dim1, dim2,
+                       ticker, interpolation){
+  validate_VolSurface(new_VolSurface(reference_date, surface_type, vol_quotes,
+                                     dim1, dim2, ticker, interpolation))
+}
+
+new_VolSurface <- function(reference_date, surface_type, vol_quotes, dim1, dim2,
+                           ticker, interpolation){
+
+  assertthat::assert_that(is.Interpolation(interpolation))
```

Comment:
why are you including an assertion in a `new_X()` method? Shouldn't this what the `validate_X()` function is there for?

Reply:
Here I am using the same logic used in ZeroCurve class where the assertion of the interpolation method is introduced in new_ZeroCurve rather than in validate_ZeroCurve. We can easily shift the assertion clause to the validation function.

Reply:
Now the assertion on the Interpolation is included in the validation function.